controller/src/main/java/io/pravega/controller/server/bucket/BucketManager.java

```diff
@@ -95,7 +102,8 @@ public void failed(State from, Throwable failure) {
 
     @Override
     protected void doStop() {
-        Futures.allOf(buckets.stream().map(bucketService -> {
+        log.info("Stop request received for bucket service {}", serviceType);
+        Futures.allOf(buckets.values().stream().map(bucketService -> {
             CompletableFuture<Void> bucketFuture = new CompletableFuture<>();
             bucketService.addListener(new Listener() {
```

Comment:
Use `Services.stopAsync` and `Services.onStop`

Reply:
I do use stopAsync. 
AbstractService does not have onStop method. And we dont want to awaitTermination so adding a listener for service state changes and completing a future.